src/BlockTypes/CustomerAccount.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * CustomerAccount class.
+ */
+class CustomerAccount extends AbstractBlock {
+	const TEXT_ONLY   = 'text_only';
+	const ICON_ONLY   = 'icon_only';
+	const DISPLAY_ALT = 'alt';
+
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'customer-account';
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 *
+	 * @return string Rendered block output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+
+		$account_link = get_option( 'woocommerce_myaccount_page_id' ) ? wc_get_account_endpoint_url( 'dashboard' ) : wp_login_url();
+
+		return "<div class='wp-block-woocommerce-customer-account " . esc_attr( $classes_and_styles['classes'] ) . "' style='" . esc_attr( $classes_and_styles['styles'] ) . "'>
+			<a href='$account_link'>
+				{$this->icon( $attributes )}<span class='label'>{$this->label( $attributes )}</span>
+			</a>
+		</div>";
+	}
+
+	/**
+	 * Gets the icon to render depending on the iconStyle and displayStyle.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return string Label to render on the block
+	 */
+	private function icon( $attributes ) {
```

Comment:
Nit: I would try to use a verb in the method name. Something like `get_icon_markup()` or `render_icon()`.